core/src/Xios.cpp

```diff
@@ -213,69 +275,517 @@ void Xios::setCalendarTimestep(cxios_duration timestep)
+ * Set the local longitude size for a given domain
  *
- * @param duration
+ * @param the domain ID
+ * @param the local longitude size
  */
-void Xios::printCXiosDuration(cxios_duration duration)
+void Xios::setDomainLongitudeSize(std::string domainId, int size)
 {
-    std::cout << " year     " << duration.year << std::endl;
-    std::cout << " month    " << duration.month << std::endl;
-    std::cout << " day      " << duration.day << std::endl;
-    std::cout << " hour     " << duration.hour << std::endl;
-    std::cout << " minute   " << duration.minute << std::endl;
-    std::cout << " second   " << duration.second << std::endl;
-    std::cout << " timestep " << duration.timestep << std::endl;
+    xios::CDomain* domain = getDomain(domainId);
+    cxios_set_domain_ni(domain, size);
+}
+
+/*!
+ * Set the local latitude size for a given domain
+ *
+ * @param the domain ID
+ * @param the local longitude size
+ */
+void Xios::setDomainLatitudeSize(std::string domainId, int size)
+{
+    xios::CDomain* domain = getDomain(domainId);
+    cxios_set_domain_nj(domain, size);
+}
+
+/*!
+ * Set the local start longitude for a given domain
+ *
+ * @param the domain ID
+ * @return the local start longitude
+ */
+void Xios::setDomainLongitudeStart(std::string domainId, int start)
+{
+    xios::CDomain* domain = getDomain(domainId);
+    cxios_set_domain_ibegin(domain, start);
+}
+
+/*!
+ * Set the local start latitude for a given domain
+ *
+ * @param the domain ID
+ * @return the local start latitude
+ */
+void Xios::setDomainLatitudeStart(std::string domainId, int start)
+{
+    xios::CDomain* domain = getDomain(domainId);
+    cxios_set_domain_jbegin(domain, start);
+}
+
+/*!
+ * Set the local longitude values for a given domain
+ *
+ * @param the domain ID
+ * @return the local longitude values
+ */
+void Xios::setDomainLongitudeValues(std::string domainId, std::vector<double> values)
+{
+    int size = getDomainLongitudeSize(domainId);
+    xios::CDomain* domain = getDomain(domainId);
+    cxios_set_domain_lonvalue_1d(domain, values.data(), &size);
+}
+
+/*!
+ * Set the local latitude values for a given domain
+ *
+ * @param the domain ID
+ * @return the local latitude values
+ */
+void Xios::setDomainLatitudeValues(std::string domainId, std::vector<double> values)
+{
+    int size = getDomainLatitudeSize(domainId);
+    xios::CDomain* domain = getDomain(domainId);
+    cxios_set_domain_latvalue_1d(domain, values.data(), &size);
+}
+
+/*!
+ * Get the domain associated with a given ID
+ *
+ * @param the domain ID
+ * @return a pointer to the XIOS CDomain object
+ */
+xios::CDomain* Xios::getDomain(std::string domainId)
+{
+    xios::CDomain* domain = NULL;
+    cxios_domain_handle_create(&domain, domainId.c_str(), domainId.length());
+    return domain;
+}
+
+/*!
+ * Get the type of a given domain
+ *
+ * @param the domain ID
+ * @return the corresponding domain type
+ */
+std::string Xios::getDomainType(std::string domainId)
+{
+    int size = 20;
```

Comment:
This is just the string length used in the toy code. We'd probably want to use a larger value.

Reply:
we should check what happens if `domain` is larger than 20. Does `xios` throw an error?

Reply:
I tried this with `size = 40` and the test still passed. Although, just to note, the size here is string length rather than domain size. So the `size` here is now set across the whole class in f60e88fc19cc50ff0cbde3b0438daa6798b5d6ef.